routs/soldiers.js

```diff
@@ -0,0 +1,29 @@
+const soldiers = require('../collections/soldiers.js');
+const urlib = require('url');
+
+function handleSoldierReq(req, data, done) {
+
+  const parsedUrlString = req.url.split("/");
+  if (parsedUrlString.length >= 3) {
+    done("invalid url request");
+  }
+  const urlParams = parsedUrlString[2];
+  const parsedUrl = new urlib.URL("http://localhost:3000/" + req.url);
+  if (req.method === "GET") {
+    if (parsedUrlString.length === 3 && /tt[0-9]{7}/.test(urlParams)) {
+      soldiers.findSoldier(urlParams , null, done)
+    } else if (parsedUrl.searchParams.has("name")) {
+      soldiers.findSoldier(null, parsedUrl.searchParams.get("name"), done)
+    } else if (parsedUrlString.length === 2) {
+      soldiers.findSoldier(null, null, done)
+    } else {
+      done("invalid url request")
+    }
+  } else if (req.method === "POST" && parsedUrlString.length === 2) {
+    soldiers.insertSoldier(data, done);
+  } else {
+    done("invalid url request")
```

Comment:
There are standard error that are returned in such cases, e.g. 404. Or other options which you find fitting for the situation.
You don't need to invent new error messages yourself and more importantly return an informative status code.